### ISSUE TITLE
LazyJsonLineReader creates null column if single value is null

### ISSUE THREAD

**Issue by evbo** [pola-rs/polars#13183]
There seems to have been a regression @evbo - it's probably simpler to create a new issue.

In py-polars, it appears to have last worked in `0.19.15` 

```python
pl.__version__
# '0.19.15'

pl.read_ndjson(b"""
{"map": "a", "start_time": 0.795, "end_time": 1.495}
{"map": "a", "start_time": 1.6239999999999999, "end_time": 2.0540000000000003}
{"map": "c", "start_time": 2.184, "end_time": 2.645}
{"map": "a", "start_time": null, "end_time": null}
""".strip())

# shape: (4, 3)
# ┌─────┬────────────┬──────────┐
# │ map ┆ start_time ┆ end_time │
# │ --- ┆ ---        ┆ ---      │
# │ str ┆ f64        ┆ f64      │
# ╞═════╪════════════╪══════════╡
# │ a   ┆ 0.795      ┆ 1.495    │
# │ a   ┆ 1.624      ┆ 2.054    │
# │ c   ┆ 2.184      ┆ 2.645    │
# │ a   ┆ null       ┆ null     │
# └─────┴────────────┴──────────┘
```

```python
pl.__version__
# '0.19.17'

# shape: (4, 3)
# ┌─────┬────────────┬──────────┐
# │ map ┆ start_time ┆ end_time │
# │ --- ┆ ---        ┆ ---      │
# │ str ┆ str        ┆ str      │
# ╞═════╪════════════╪══════════╡
# │ a   ┆ null       ┆ null     │
# │ a   ┆ null       ┆ null     │
# │ c   ┆ null       ┆ null     │
# │ a   ┆ null       ┆ null     │
# └─────┴────────────┴──────────┘
```

_Originally posted by @cmdlineluser in https://github.com/pola-rs/polars/issues/5013#issuecomment-1866910590_

**Comment by cmdlineluser** [comment]
Looks like it changed after https://github.com/pola-rs/polars/pull/12677

**Comment by ritchie46** [comment]
Will take a look

**Comment by cmdlineluser** [comment]
This has been fixed by #13206
